src/middleware/authorization.ts

```diff
@@ -0,0 +1,16 @@
+import 'dotenv/config';
+import { Request, Response, NextFunction } from 'express';
+import { CurrentUser } from '../types';
+
+export async function isAdmin(req: Request, res: Response, next: NextFunction) {
+  const currentUser : CurrentUser = req.user;
+
+  if(currentUser.role !== 'admin') {
+    return res.status(403).send({
+        "data": null,
+        "error": {
+          "message": "You must be authorized user"
+        }
+      });
+  }
+}
```

Comment:
W: This is just an alternative solution using a [closure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures). By using this you can move the x-user-id validation to the middleware and avoid duplicating this logic in case you need to use it for other routes:
```suggestion
export function isAdmin({ shouldValidateUser }: Options = {}) {
  return (req: Request, res: Response, next: NextFunction) => {
    const currentUser: CurrentUser = req.user;

    if (currentUser.role !== "admin") {
      return res.status(403).send({
        data: null,
        error: {
          message: "You must be authorized user",
        },
      });
    }

    if (shouldValidateUser && !req.get("x-user-id")) {
      return res.status(400).send({
        data: null,
        error: {
          message:
            "x-user-id header is missing, unable to complete the request",
        },
      });
    }
  };
}
```

Reply:
I just found an small bug. You need to call the next function after the validation to allow the request to reach the controller handler